plugins/serializers/graphite/graphite.go

```diff
@@ -308,11 +316,18 @@ func buildTags(tags map[string]string) string {
 	return tagStr
 }
 
-func sanitize(value string) string {
+func strictSanitize(value string) string {
 	// Apply special hyphenation rules to preserve backwards compatibility
 	value = hyphenChars.Replace(value)
 	// Apply rule to drop some chars to preserve backwards compatibility
 	value = dropChars.Replace(value)
 	// Replace any remaining illegal chars
-	return allowedChars.ReplaceAllLiteralString(value, "_")
+	return strictAllowedChars.ReplaceAllLiteralString(value, "_")
+}
+
+func compatibleSanitize(name string, value string) string {
+	name = compatibleAllowedCharsName.ReplaceAllLiteralString(name, "_")
+	value = compatibleAllowedCharsValue.ReplaceAllLiteralString(value, "_")
+	value = regexp.MustCompile(`^[~]*(.*)`).FindStringSubmatch(value)[1]
```

Comment:
please move this out to a variable so it gets checked earlier, is named, and is reused. I think it's also possible for this function to not return the match you're expecting, so maybe we can't assume the `[1]` match exists?

Reply:
Yeah, the regexes are a bit dense, but it's the only way I can think to effieciently exclude the disallowed characters.
You'd like the last regex (checking for leading tilde) to be broken out to a variable? 
I've tested the function with inputs where the value has, and doesn't have, tildes and it seems to always return what I expect. I'll try some more to see if it works properly.

Reply:
I [tested](https://play.golang.org/p/E-loRxX2Zeo) a whole pile of possible values that might break the existance of the `[1]`, and I wasn't able to do so in any case. Even providing an empty value doesn't do it, probably because the matching group is so open, that it accepts anything.